### ISSUE TITLE
Issue while Linking CXX shared library libassimp.so

### ISSUE THREAD

**Issue by feiyuren233** [assimp/assimp#746]
I'm got this error when I was building the 3.2.0 version on Ubuntu.

    /usr/bin/ld: ../contrib/zlib/libzlibstatic.a(compress.o): relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC
    ../contrib/zlib/libzlibstatic.a: error adding symbols: Bad value

But I could build it successfully under another system
Anyone know what is the cause of this issue and how to fix it?
Thanks in advance!

**Comment by kimkulling** [comment]
Which version of Ubuntu + gcc are you using?

**Comment by feiyuren233** [comment]
Thanks for replying.
It seems that it works on Debian gcc 4.9.2 but not on Ubuntu gcc 4.8.4.

**Comment by 0x2A** [comment]
Just try to recompile with `-fPIC` flag. See this post for more info: http://stackoverflow.com/questions/19364969/compilation-fails-with-relocation-r-x86-64-32-against-rodata-str1-8-can-not

**Comment by feiyuren233** [comment]
Oh yeah it worked with the -fPIC flag, my bad to miss the info given by gcc. thank you guys for replying!

**Comment by kankaristo** [comment]
How do you specify the `-fPIC` flag? I've tried a few different ways, but still get the same error.

Also, shouldn't the flag be set by CMake, instead of setting it manually?

I'm thinking that the default build commands should "just work"?
```sh
cmake -G 'Unix Makefiles' -DASSIMP_ENABLE_BOOST_WORKAROUND=ON
make #-j8
```

I'm on Ubuntu 15.10 with GCC 5.2.1 (Assimp 3.2).

**Comment by kankaristo** [comment]
Ah, I got it working with:
```sh
cmake -G 'Unix Makefiles' -DASSIMP_ENABLE_BOOST_WORKAROUND=ON -DCMAKE_C_FLAGS=-fPIC -DCMAKE_CXX_FLAGS=-fPIC
make #-j8
```

Although `-fPIC` should probably be enabled by default, since the build is broken without it.

The build still fails saying something about `gtest`, which seems to have been fixed in https://github.com/assimp/assimp/pull/783. The Git version builds fine (after adding `-fPIC`).

Version 3.2 seems to be broken (on Ubuntu at least), maybe release version 3.2.1?
